subsys/bluetooth/controller/ll_sw/lll_filter.h

```diff
@@ -4,27 +4,31 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_BT_CTLR_FAL_SIZE)
-#define FAL_SIZE CONFIG_BT_CTLR_FAL_SIZE
+#if defined(CONFIG_BT_CTLR_FAL_SIZE) && defined(CONFIG_BT_CTLR_RL_SIZE)
+#define LLL_FILTER_SIZE MAX(CONFIG_BT_CTLR_FAL_SIZE, CONFIG_BT_CTLR_RL_SIZE)
+#elif defined(CONFIG_BT_CTLR_FAL_SIZE)
+#define LLL_FILTER_SIZE CONFIG_BT_CTLR_FAL_SIZE
+#elif defined(CONFIG_BT_CTLR_RL_SIZE)
+#define LLL_FILTER_SIZE CONFIG_BT_CTLR_RL_SIZE
 #else
-#define FAL_SIZE 8
-#endif /* CONFIG_BT_CTLR_FAL_SIZE */
+#define LLL_FILTER_SIZE 8
+#endif /* CONFIG_BT_CTLR_FAL_SIZE && CONFIG_BT_CTLR_RL_SIZE */
 
 #define IRK_SIZE               16
 #define FILTER_IDX_NONE        0xFF
-#define LLL_FILTER_BITMASK_ALL (BIT(FAL_SIZE) - 1)
+#define LLL_FILTER_BITMASK_ALL (BIT(LLL_FILTER_SIZE) - 1)
 
 struct lll_filter {
-#if (FAL_SIZE <= 8)
+#if (LLL_FILTER_SIZE <= 8)
 	uint8_t  enable_bitmask;
 	uint8_t  addr_type_bitmask;
-#elif (FAL_SIZE <= 16)
+#elif (LLL_FILTER_SIZE <= 16)
```

Comment:
```suggestion
#if (LLL_FILTER_SIZE <= USHRT_WIDTH)
```
Add macro `#define USHRT_WIDTH (sizeof(uint16_t) * CHAR_BIT)`. The macro is provided in C23 standard (ref: https://en.cppreference.com/w/c/types/limits).

Reply:
Instead of overcomplicating, please use:
```
#define LLL_FILTER_BITMASK8_COUNT 8
#define LLL_FILTER_BITMASK16_COUNT 16
```

Reply:
We already discussed offline with @ppryga. Would this be ok for you if I use: `(sizeof(uint16_t) * CHAR_BIT)` ?
Thanks to that there would be no need to introduce additional define

Reply:
`(sizeof(uint16_t) * CHAR_BIT)` does not convey that it is resolving/filter list max size. I prefer to use a define which can then use `__CHAR_BIT__` like done here: https://github.com/zephyrproject-rtos/zephyr/blob/d2a56c5047f9e7f10b1b0df53d57917d3980bf72/include/sys/util.h#L47-L52

i.e.:
```
#define LLL_FILTER_SIZE_PER_BITMASK8  (sizeof(uint8_t) * __CHAR_BIT__)
#define LLL_FILTER_SIZE_PER_BITMASK16 (sizeof(uint16_t) * __CHAR_BIT__)
```

@mtpr-ot @wopu-ot does your platform support `__CHAR_BIT__` ?

Reply:
`CHAR_BIT` is defined by C standard, it should be available for all platform.
I'm fine with current implementation. In future is should be changed to macro provided by C23 standart.

Reply:
@MarekPieta are you going to apply comment from @cvinayak ?